### PR TITLE
Add header files to CMakeLists.txt files

### PR DESCRIPTION
**Pull request by erlliam**

### CODE REVIEW

**Comment by strager** [comment]
An explanation would be nice to include in the commit message.

Should we also list header files for quick-lint-js-test and for quick-lint-js-benchmark-support?

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -69,6 +69,58 @@ quick_lint_js_add_library(
   vector.cpp
   vim-qflist-json-error-reporter.cpp
   wasm-demo-error-reporter.cpp
+  quick-lint-js/assert.h
```
Comment (strager) [inline src/CMakeLists.txt:72]: Nit: I'd like this list to be sorted lexicographically, such that `quick-lint-js/assert.h` appears between `parse.cpp` and `text-error-reporter.cpp`.

**Review by strager — APPROVED** [review]

**Comment by erlliam** [comment]
It should be good to go.

**Comment by strager** [comment]
Landed as commit 0f6cfbdcd. Thanks!

**Pull request closed without merging**